build.sh

```diff
@@ -236,27 +239,60 @@ function build_qemu {
 }
 
 function build_odroid_h4 {
-  DEFCONFIG="configs/config.hardkernel_odroid_h4"
+  DEFCONFIG=$1
   FW_VERSION=$(cat ${DEFCONFIG} | grep CONFIG_LOCALVERSION | cut -d '=' -f 2 | tr -d '"')
 
   # checkout several submodules needed by these boards (some others are checked
   # out by coreboot's Makefile)
   git submodule update --init --force --checkout \
       3rdparty/dasharo-blobs
 
-  docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
-    -v $HOME/.ssh:/home/coreboot/.ssh \
-    -w /home/coreboot/coreboot coreboot/coreboot-sdk:$SDKVER \
-    /bin/bash -c "make distclean"
+  if [ "${AIRGAP}" -eq 1 ]; then
+    docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
+      -v $HOME/.ssh:/home/coreboot/.ssh \
+      -w /home/coreboot/coreboot ${DASHARO_SDK} \
+      /bin/bash -c "make clean"
+  else
+    docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
+      -v $HOME/.ssh:/home/coreboot/.ssh \
+      -w /home/coreboot/coreboot ${DASHARO_SDK} \
+      /bin/bash -c "make distclean"
+  fi
 
   cp $DEFCONFIG .config
 
   echo "Building Dasharo compatbile with Hardkernel ODROID H4 (version $FW_VERSION)"
 
-  docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
-    -v $HOME/.ssh:/home/coreboot/.ssh \
-    -w /home/coreboot/coreboot coreboot/coreboot-sdk:$SDKVER \
-    /bin/bash -c "make olddefconfig && make -j$(nproc)"
+  if [ "${AIRGAP}" -eq 1 ]; then
+
+    # In this situation we assume that provided repository is ready to be used
+    # and nothing should be downloaded during build process.
+
+    if [ -d "${EDK2_REPO_PATH}" ]; then
+      # Without following sequence workspce would be created by docker with root
+      # privilidges and build will fail.
+      # Target directory
+      TARGET_DIR="payloads/external/edk2/workspace/Dasharo"
+      mkdir -p "$TARGET_DIR"
+      chown -R $(id -u):$(id -g) "$TARGET_DIR"
+      chmod -R 755 "$TARGET_DIR"
+      docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
+        -v $HOME/.ssh:/home/coreboot/.ssh \
+        --network none \
+        ${EDK2_REPO_PATH:+-v $EDK2_REPO_PATH:/home/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo} \
+        -e BUILD_TIMELESS=${BUILD_TIMELESS} \
+        -w /home/coreboot/coreboot ${DASHARO_SDK} \
+        /bin/bash -c "make olddefconfig && make -j$(nproc)"
+    else
+      echo "EDK2_REPO_PATH is not defined in AIRGAP!"
+      exit 1
+    fi
+  else
+    docker run --rm -t -u $UID -v $PWD:/home/coreboot/coreboot \
+      -v $HOME/.ssh:/home/coreboot/.ssh \
+      -w /home/coreboot/coreboot ${DASHARO_SDK} \
+      /bin/bash -c "make olddefconfig && make -j$(nproc)"
+  fi
```

Comment:
Should be applied to all boards, not just selected ones. Maybe it would also be worth to separate it into a function to reduce the LOC and overall boilerplate. I think the docker commands to start the build are the same.